docs/hr/content/docs/reusable_snippets/insert_data.md

```diff
@@ -13,29 +13,29 @@ In order to create data, we need to create a `Schema` for encoding our special `
 <Tabs>
     <TabItem value="MongoDB" label="MongoDB" default>
         ```python
-        from superduperdb import Document
+        from superduperdb import Document, DataType
         
-        def do_insert(data):
-            schema = None
-            
+        def do_insert(data, schema = None):
             
-            if schema is None and (datatype is None  or isinstance(datatype, str)) :
-                data = [Document({'x': x}) for x in data]
+            if schema is None and (datatype is None or isinstance(datatype, str)):
+                data = [Document({'x': x['x'], 'y': x['y']}) for x in data]
                 db.execute(table_or_collection.insert_many(data))
-            elif schema is None and datatype is not None and isintance():
-                data = [Document({'x': datatype(x)}) for x in data]
+            elif schema is None and datatype is not None and isinstance(datatype, DataType):
+                data = [Document({'x': datatype(x['x']), 'y': x['y']}) for x in data]
                 db.execute(table_or_collection.insert_many(data))
             else:
-                data = [Document({'x': x}) for x in data]
-                db.execute(table_or_collection.insert_many(data, schema='my_schema'))        
+                data = [Document({'x': x['x'], 'y': x['y']}) for x in data]
+                db.execute(table_or_collection.insert_many(data, schema=schema))
+        
```

Comment:
If the data format is non-generic or if the insert operation cannot be generic, it is recommended to create a new snippet or create a cell in the use case template to insert data.